components/Modal/Portal.tsx

```diff
@@ -0,0 +1,74 @@
+import { createRef, ReactElement, useEffect, useState } from 'react';
```

Comment:
This whole file creates a new reusable component that creates a React Portal for the modals.
With some work, can be made a generic Portal creator for UiKit.